packages/opentelemetry-shim-opentracing/src/shim.ts

```diff
@@ -0,0 +1,284 @@
+import * as opentracing from 'opentracing';
+
+function translateReferences(
+  references: opentracing.Reference[]
+): types.Link[] {
+  const links: types.Link[] = [];
+  for (const reference of references) {
+    const context = reference.referencedContext();
+    if (context instanceof SpanContextShim) {
+      links.push({
+        spanContext: (context as SpanContextShim).getSpanContext(),
+        attributes: { 'span.kind': reference.type },
+      });
+    }
+  }
+  return links;
+}
+
+function translateSpanOptions(
+  options: opentracing.SpanOptions
+): types.SpanOptions {
+  const opts: types.SpanOptions = {
+    startTime: options.startTime,
+  };
+
+  // because there's no `Links` in SpanOptions, we set them in `TracerShim.startSpan()`
+  if (options.childOf) {
+    if (options.childOf instanceof SpanShim) {
+      opts.parent = (options.childOf as SpanShim).getSpan();
+    } else if (options.childOf instanceof SpanContextShim) {
+      opts.parent = (options.childOf as SpanContextShim).getSpanContext();
+    }
+  }
+  return opts;
+}
+
+/**
+ * SpanContextShim wraps a {@link types.SpanContext} and implements the
+ * OpenTracing span context API.
+ */
+export class SpanContextShim extends opentracing.SpanContext {
+  private readonly _spanContext: types.SpanContext;
+
+  constructor(spanContext: types.SpanContext) {
+    super();
+    this._spanContext = spanContext;
+  }
+
+  /**
+   * Returns the underlying {@link types.SpanContext}
+   */
+  getSpanContext(): types.SpanContext {
+    return this._spanContext;
+  }
+
+  /**
+   * Returns the trace ID as a string.
+   */
+  toTraceId(): string {
+    return this._spanContext.traceId;
+  }
+
+  /**
+   * Returns the span ID as a string.
+   */
+  toSpanId(): string {
+    return this._spanContext.spanId;
+  }
+}
+
+/**
+ * TracerShim wraps a {@link types.Tracer} and implements the
+ * OpenTracing tracer API.
+ */
+export class TracerShim extends opentracing.Tracer {
+  private readonly _tracer: types.Tracer;
+  constructor(tracer: types.Tracer) {
+    super();
+    this._tracer = tracer;
+  }
+
+  startSpan(
+    name: string,
+    options: opentracing.SpanOptions = {}
+  ): opentracing.Span {
+    const span = this._tracer.startSpan(name, translateSpanOptions(options));
+
+    if (options.tags) {
+      span.setAttributes(options.tags);
+    }
+
+    if (options.references) {
+      const links = translateReferences(options.references);
+      for (const link of links) {
+        span.addLink(link.spanContext, link.attributes);
+      }
+    }
+
+    return new SpanShim(this, span);
+  }
+
+  _inject(
+    spanContext: opentracing.SpanContext,
+    format: string,
+    carrier: unknown
+  ): void {
+    const opentelemSpanContext: types.SpanContext = (spanContext as SpanContextShim).getSpanContext();
+    switch (format) {
+      // tslint:disable-next-line:no-switch-case-fall-through
+      case opentracing.FORMAT_HTTP_HEADERS:
+      case opentracing.FORMAT_TEXT_MAP:
```

Comment:
I'm not sure that opentracing.FORMAT_HTTP_HEADERS is the same as opentracing.FORMAT_TEXT_MAP

Reply:
since opentelemetry doesn't support a text map format, http is the closest we get. fwiw both http and text have the same carrier type in opentracing and http is a stricter subset of text map so I think they're basically equivalent, plus it's going to the same inject/extract implementation